### PR TITLE
Answers -> Search

### PR DESCRIPTION
**Pull request by ElemelonWind**
J=SLAP-2196
TEST=auto

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,114 +1,114 @@
...
 }
 ```
 
-## `AnswersHeadlessContext`
-### Class Components
+## `SearchHeadlessContext`
+## Class Components
 
-For users that want to use class components instead of functional components, you can use the `AnswersHeadlessContext` directly to dispatch actions and receive updates from state.
+For users that want to use class components instead of functional components, you can use the `SearchHeadlessContext` directly to dispatch actions, and the `subscribeToStateUpdates` HOC to receive updates from state.
```
Comment (nmanu1) [inline README.md:96]: `subscribeToStateUpdates` was recently deprecated, so it looks like some changes in `develop` were reverted. this feature branch may need to be rebased off of the latest `develop`

File: README.md [hunk 2]
```diff
@@ -1,114 +1,114 @@
...
     }
-  }, [answers]);
+  }, [search]);
 
   return <input onChange={handleTyping} onKeyDown={handleKeyDown}/>;
 }
 ```
 
-## `AnswersHeadlessContext`
-### Class Components
```
Comment (nmanu1) [inline README.md:94]: was this formatting change intentional?

File: README.md [hunk 3]
```diff
@@ -1,114 +1,114 @@
...
+## Respond to State Updates with `useSearchState`
 
-`useAnswersState` reads a value from the `AnswersHeadless` state and subscribes to updates.
+`useSearchState` reads a value from the `SearchHeadless` state and subscribes to updates.
 
 ```tsx
-import { useAnswersState } from '@yext/answers-headless-react';
+import { useSearchState } from '@yext/search-headless-react';
 
-export default function MostRecentSearch() {
```
Comment (nmanu1) [inline README.md:59]: can we add `export` back in? otherwise `default` should also be removed because it can't be by itself

Comment (ElemelonWind) [reply README.md:59]: I probably randomly pressed backspace for both of these whoops

File: src/deprecated.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+/**
+ * @deprecated useAnswersState has been deprecated and replaced by useSearchState
+ */
+export function useAnswersState<T>(stateSelector: StateSelector<T>): T {
+  return useSearchState(stateSelector);
+}
+
+/**
+ * @deprecated useAnswersUtilities has been deprecated and repalced by useSearchUtilities
```
Comment (nmanu1) [inline src/deprecated.ts:42]: nit: "replaced"

File: src/deprecated.ts [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+ */
+export function useAnswersUtilities(): SearchUtilities { return useSearchUtilities(); }
+
+/**
+ * @deprecated AnswersHeadlessProvider has been deprecated and replaced by SearfchHeadlessProvider
```
Comment (nmanu1) [inline src/deprecated.ts:47]: nit: "SearchHeadlessProvider"

File: src/subscribeToStateUpdates.tsx [hunk 6]
```diff
@@ -1,32 +1,32 @@
...
 /**
  * Generates a HOC that updates a given Component's props based on the current
- * answers-headless state and a given mapping function.
+ * search-headless state and a given mapping function.
  *
  * @deprecated
- * For class component, use `AnswersHeadlessContext` directly to dispatch actions and receive state updates.
- * For functional component, use `useAnswersActions` and `useAnswersState` instead.
+ * For class component, use `SearchHeadlessContext` directly to dispatch actions and receive state updates.
+ * For functional component, use `useSearchActions` and `useSearchState` instead.
```
Comment (nmanu1) [inline src/subscribeToStateUpdates.tsx:15]: nit: can we make "class component" and "functional component" plural here

**Review by nmanu1 — APPROVED** [review]

**Pull request merged**